src/unity/toolkits/object_detection/object_detector.cpp

```diff
@@ -469,10 +487,20 @@ variant_map_type object_detector::evaluate(
     return std::accumulate(v.begin(), v.end(), 0.f) / v.size();
   };
   variant_map_type result_map;
-  result_map["average_precision_50"] = create_dict(average_precision_50);
-  result_map["average_precision"] = create_dict(average_precision);
-  result_map["mean_average_precision_50"] = compute_mean(average_precision_50);
-  result_map["mean_average_precision"] = compute_mean(average_precision);
+  
+  if (std::find(metrics.begin(), metrics.end(), AP) != metrics.end() ){
+    result_map["average_precision"] = create_dict(average_precision);
```

Comment:
We may as well use the new constants you defined here, instead of these string literals. So `AP` instead of `"average_precision"`